### PR TITLE
ApiClient upgrade to return SigmfMetaData and unit tests

### PR DESCRIPTION
**Pull request by PaulDFoster**
### What does this PR do?
Tests for datasources\ApiClient.ts methods
Upgrade to metadata\ApiClient.ts methods to return a SigmfMetaData object
Tests for metadata\ApiClient.ts methods

Tests implemented in appropriate __tests__ directories local to the .ts file being tested
Tests using nock to mock the api methods as previously used in the project codebase

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you successfully ran tests with your changes locally? Including container validation.
* [X] Have you lint your code locally prior to submission?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [X] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: client/src/api/datasource/__tests__/ApiClient.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    account: 'gnuradio',
+    container: 'iqengine',
+    description: 'GNU Radio hosted test data',
+  };
+  nock('http://localhost:3000').get('/api/datasources/gnuradio/iqengine').reply(200, data);
+
+  const client = new ApiClient();
+  const result = await client.get('gnuradio', 'iqengine');
+  expect(result).toEqual(data);
+});
```
Comment (Nepomuceno) [inline client/src/api/datasource/__tests__/ApiClient.test.tsx:43]: I would add a describe as a way to group those tests so we know they are part of the same class that we are testing.

File: client/src/api/metadata/ApiClient.ts [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
   }
+
   async getDataSourceMeta(account: string, container: string): Promise<SigMFMetadata[]> {
     const response = await axios.get(`/api/datasources/${account}/${container}/meta`);
-    return response.data;
+    let objectFromJson = response.data;
+    let responseMetaData: SigMFMetadata[] = objectFromJson.map((meta) => {
+      let newMeta = Object.assign(new SigMFMetadata(), meta);
+      if (newMeta.annotations) {
+        newMeta.annotations = newMeta.annotations.map((annotation) => Object.assign(new Annotation(), annotation));
```
Comment (Nepomuceno) [inline client/src/api/metadata/ApiClient.ts:25]: Just a small suggestion but you could avoid the need for the check and the nasted by making the map call 
```suggestion
        newMeta.annotations = newMeta.annotations?.map((annotation) => Object.assign(new Annotation(), annotation));
```

File: client/src/api/metadata/ApiClient.ts [hunk 3]
```diff
@@ -1,18 +1,46 @@
...
+      return newMeta;
+    });
+    return responseMetaData;
   }
   async updateMeta(account: string, container: string, filePath: string, meta: SigMFMetadata): Promise<SigMFMetadata> {
     const response = await axios.put(`/api/datasources/${account}/${container}/${filePath}`, meta);
-    return response.data;
+    let responseMetaData: SigMFMetadata | null = null;
+    responseMetaData = Object.assign(new SigMFMetadata(), response.data);
+    responseMetaData.annotations = responseMetaData.annotations.map((annotation) =>
```
Comment (Nepomuceno) [inline client/src/api/metadata/ApiClient.ts:38]: Same forr this one here adding the ? could safeguard this.

File: client/src/api/metadata/__tests__/ApiClient.test.tsx [hunk 4]
```diff
@@ -0,0 +1,252 @@
+import { test, describe } from 'vitest';
+import nock from 'nock';
+import { ApiClient } from '../ApiClient';
+import { SigMFMetadata, Annotation, CaptureSegment } from '@/Utils/sigmfMetadata';
+import { json } from 'react-router-dom';
+
+const account = 'gnuradio';
+const container = 'iqengine';
+const filePath = 'bluetooth';
+
```
Comment (Jaydee123) [inline client/src/api/metadata/__tests__/ApiClient.test.tsx:10]: do we need to add a test to determine what the behaviour is when the metadata doesn't have annotations/captures?

Comment (PaulDFoster) [reply client/src/api/metadata/__tests__/ApiClient.test.tsx:10]: Added a new test

**Review by Jaydee123 — APPROVED** [review]

**Pull request merged**